spec.emu

```diff
@@ -271,6 +463,86 @@ contributors: Luca Casonato, Guy Bedford
             </tr>
           </table>
         </emu-table>
+
+        <emu-clause id="sec-LoadRequestedModules" type="concrete method">
+          <h1>
+            LoadRequestedModules (
+              optional _hostDefined_: anything,
+            ): a Promise object
+          </h1>
+          <dl class="header">
+            <dt>for</dt>
+            <dd>a Cyclic Module Record _module_</dd>
+          </dl>
+
+          <emu-clause id="sec-InnerModuleLoading" type="abstract operation">
+            <h1>
+              InnerModuleLoading (
+                _state_: a GraphLoadingState Record,
+                _module_: a Module Record,
+                <ins>optional _referrer_: a Cyclic Module Record,</ins>
+                <ins>optional _specifier_: a String,</ins>
+              ): ~unused~
+            </h1>
+            <dl class="header">
+              <dt>description</dt>
+              <dd>It is used by LoadRequestedModules to recursively perform the actual loading process for _module_'s dependency graph.</dd>
+            </dl>
+
+            <emu-alg>
+              1. Assert: _state_.[[IsLoading]] is *true*.
+              1. <ins>Let _importOnlyAsReflection_ be *false*.</ins>
+              1. <ins>If _referrer_ has been specified, then</ins>
+                1. <ins>Assert: _specifier_ has been specified.</ins>
+                1. <ins>If _referrer_.[[RequestedModuleReflections]] contains _specifier_, set _importOnlyAsReflection_ to *true*.</ins>
```

Comment:
Would it be possible to rather have this state passed as another argument to `InnerModuleLoading` instead of needing to check it explicitly here? Also to double-check, `import('specifier')` within the module doesn't hit this path right?

Reply:
Especially now that we have a single list of all the dependencies, we need to keep the check _somehwere_. Doing it here instead of passing it as a parameter to `InnerModuleLoading` avoids the need of passing it as another opaque parameter of HostLoadImportedModule.